packages/components/data-table-manager/src/data-table-manager.story.js

```diff
@@ -321,4 +324,157 @@ storiesOf('Components|DataTable', module)
+    );
+    const visibleColumns = tableData.visibleColumnKeys.map(
+      (columnKey) => mappedColumns[columnKey]
+    );
+
+    const columnsWithSelect = [
+      {
+        key: 'checkbox',
+        label: (
+          <CheckboxInput
+            isIndeterminate={isSelectColumnHeaderIndeterminate}
+            isChecked={countSelectedRows !== 0}
+            onChange={handleSelectColumnHeaderChange}
+          />
+        ),
+        shouldIgnoreRowClick: true,
+        align: 'center',
+        renderItem: (row) => (
+          <CheckboxInput
+            isChecked={getIsRowSelected(row.id)}
+            onChange={() => toggleRow(row.id)}
+          />
+        ),
+        disableResizing: true,
+      },
+      ...visibleColumns,
+    ];
+
+    const tableSettingsChangeHandler = {
+      [UPDATE_ACTIONS.COLUMNS_UPDATE]: (visibleColumnKeys) =>
+        setTableData({
+          ...tableData,
+          visibleColumnKeys,
+        }),
+      [UPDATE_ACTIONS.IS_TABLE_CONDENSED_UPDATE]: setIsCondensed,
+      [UPDATE_ACTIONS.IS_TABLE_WRAPPING_TEXT_UPDATE]: setIsWrappingText,
+    };
+
+    const displaySettingsButtons = showDisplaySettingsConfirmationButtons
+      ? {
+          primaryButton: <FooterPrimaryButton />,
+          secondaryButton: <FooterSecondaryButton />,
+        }
+      : {};
+
+    const columnManagerButtons = showColumnManagerConfirmationButtons
+      ? {
+          primaryButton: <FooterPrimaryButton />,
+          secondaryButton: <FooterSecondaryButton />,
+        }
+      : {};
+
+    const displaySettings = {
+      disableDisplaySettings: boolean('disableDisplaySettings', false),
+      isCondensed,
+      isWrappingText,
+      ...displaySettingsButtons,
+    };
+
+    const columnManager = {
+      areHiddenColumnsSearchable: boolean('areHiddenColumnsSearchable', true),
+      searchHiddenColumns: (searchTerm) => {
+        setTableData({
+          ...tableData,
+          columns: initialColumnsState.filter(
+            (column) =>
+              tableData.visibleColumnKeys.includes(column.key) ||
+              column.label
+                .toLocaleLowerCase()
+                .includes(searchTerm.toLocaleLowerCase())
+          ),
+        });
+      },
+      disableColumnManager: boolean('disableColumnManager', false),
+      visibleColumnKeys: tableData.visibleColumnKeys,
+      hideableColumns: tableData.columns,
+      ...columnManagerButtons,
+    };
+
+    return (
+      <DataTableManagerProvider
+        columns={withRowSelection ? columnsWithSelect : visibleColumns}
+        displaySettings={displaySettings}
+        onSettingsChange={(action, nextValue) => {
+          tableSettingsChangeHandler[action](nextValue);
+        }}
+        columnManager={columnManager}
+      >
+        <Spacings.Stack>
+          <DataTableManager />
+          <SearchTextInput
+            placeholder={text('placeholder', 'Dummy search component')}
+            isReadOnly
+          />
+          <DataTable
```

Comment:
I think it would be great to have the components in a more deep component tree to really showcase consumers can use them wherever they want.

Just an example:
```jsx
return (
      <DataTableManagerProvider
        columns={withRowSelection ? columnsWithSelect : visibleColumns}
        displaySettings={displaySettings}
        onSettingsChange={(action, nextValue) => {
          tableSettingsChangeHandler[action](nextValue);
        }}
        columnManager={columnManager}
      >
        <header>
          <Spacings.Inline justifyContent="flex-end">
            <DataTableManager />
          </Spacings.Inline>
          <hr style={{ marginBottom: '24px' }} />
        </header>
        <main>
          <DataTable
            rows={withRowSelection ? rowsWithSelection : rows}
            sortedBy={sortedBy}
            onSortChange={onSortChange}
            sortDirection={sortDirection}
          />
        </main>
        <br />
        <hr />
      </DataTableManagerProvider>
```

Reply:
updated here: https://github.com/commercetools/ui-kit/pull/2807/commits/d1c167920c31da23e74275d1a3aa0912d539dc2d

Thank you